textdescriptives/components/quality.py

```diff
@@ -434,36 +509,100 @@ def quality_getter(self, span: Span) -> Dict[str, Union[float, int, bool]]:
         quality = {}
         for name, getter in self.getters.items():
             if name == "top_ngram_chr_fraction":
-                chr_frac = getter(span)
+                chr_frac = getter(span)  # type: ignore
                 for n_gram, frac in chr_frac.items():
                     quality[f"top_{n_gram}-gram_chr_fraction"] = frac
             elif name == "duplicate_ngram_chr_fraction":
-                chr_frac = getter(span)
+                chr_frac = getter(span)  # type: ignore
                 for n_gram, frac in chr_frac.items():
                     quality[f"duplicate_{n_gram}-gram_chr_fraction"] = frac
             else:
-                quality[name] = getter(span)
+                quality[name] = getter(span)  # type: ignore
         return quality
 
+    @staticmethod
+    def is_within_range(rangetuple: Interval, value: float) -> bool:
+        """Check if a value is within a range tuple. If one of the values in
+        the range tuple is None it is considered to be unbounded.
+
+        Args:
+            rangetuple (RangeTuple): range tuple
+            value (float): value to check
+
+        Returns:
+            bool: True if value is within range
+        """
+        return (rangetuple[0] is None or rangetuple[0] <= value) and (
+            rangetuple[1] is None or value <= rangetuple[1]
+        )
+
     def passed_quality_thresholds(self, span: Span) -> bool:
         """Checks whether a span passed the quality thresholds."""
         quality = span._.quality
-        for name, threshold in self.quality_thresholds.items():
-            if name not in quality:
-                raise KeyError(f"Quality metric {name} not found in doc._.quality")
-            if isinstance(threshold, bool):
-                if quality[name] != threshold:
+        qt = self.quality_thresholds
+
+        # heuristic quality filters
+        if not self.is_within_range(qt.n_stop_words, quality["n_stop_words"]):
+            return False
+        if not self.is_within_range(qt.alpha_ratio, quality["alpha_ratio"]):
+            return False
+        if not self.is_within_range(qt.mean_word_length, quality["mean_word_length"]):
+            return False
+        if not self.is_within_range(qt.doc_length, quality["doc_length"]):
+            return False
+        if not self.is_within_range(
+            qt.proportion_ellipsis,
+            quality["proportion_ellipsis"],
+        ):
```

Comment:
I feel like we could loop this.. zip the qt.X and keys?

Reply:
We could. I simply liked that it was explicit about how each key is treated.

Reply:
No strong opinion on my end, let's keep it as is